tests/tx/test_verification.py

```diff
@@ -0,0 +1,448 @@
+            patch.object(Transaction, 'verify_weight', verify_weight_wrapped),
+            patch.object(Transaction, 'verify_pow', verify_pow_wrapped),
+            patch.object(Transaction, 'verify_number_of_inputs', verify_number_of_inputs_wrapped),
+            patch.object(Transaction, 'verify_outputs', verify_outputs_wrapped),
+            patch.object(Transaction, 'verify_sigops_output', verify_sigops_output_wrapped),
+        ):
+            self.manager.verification_service.verify_basic(tx)
+
+        # Transaction methods
+        verify_parents_basic_wrapped.assert_called_once()
+        verify_weight_wrapped.assert_called_once()
+        verify_pow_wrapped.assert_called_once()
+        verify_number_of_inputs_wrapped.assert_called_once()
+        verify_outputs_wrapped.assert_called_once()
+        verify_sigops_output_wrapped.assert_called_once()
+
+    def test_transaction_verify_without_storage(self) -> None:
+        tx = self._get_valid_tx()
+
+        verify_pow_wrapped = Mock(wraps=tx.verify_pow)
+        verify_number_of_inputs_wrapped = Mock(wraps=tx.verify_number_of_inputs)
+        verify_outputs_wrapped = Mock(wraps=tx.verify_outputs)
+        verify_sigops_output_wrapped = Mock(wraps=tx.verify_sigops_output)
+
+        with (
+            patch.object(Transaction, 'verify_pow', verify_pow_wrapped),
+            patch.object(Transaction, 'verify_number_of_inputs', verify_number_of_inputs_wrapped),
+            patch.object(Transaction, 'verify_outputs', verify_outputs_wrapped),
+            patch.object(Transaction, 'verify_sigops_output', verify_sigops_output_wrapped),
+        ):
+            tx.verify_without_storage()
+
+        # Transaction methods
+        verify_pow_wrapped.assert_called_once()
+        verify_number_of_inputs_wrapped.assert_called_once()
+        verify_outputs_wrapped.assert_called_once()
+        verify_sigops_output_wrapped.assert_called_once()
+
+    def test_transaction_verify(self) -> None:
+        add_blocks_unlock_reward(self.manager)
+        tx = self._get_valid_tx()
+
+        verify_pow_wrapped = Mock(wraps=tx.verify_pow)
+        verify_number_of_inputs_wrapped = Mock(wraps=tx.verify_number_of_inputs)
+        verify_outputs_wrapped = Mock(wraps=tx.verify_outputs)
+        verify_sigops_output_wrapped = Mock(wraps=tx.verify_sigops_output)
+        verify_sigops_input_wrapped = Mock(wraps=tx.verify_sigops_input)
+        verify_inputs_wrapped = Mock(wraps=tx.verify_inputs)
+        verify_parents_wrapped = Mock(wraps=tx.verify_parents)
+        verify_sum_wrapped = Mock(wraps=tx.verify_sum)
+        verify_reward_locked_wrapped = Mock(wraps=tx.verify_reward_locked)
+
+        with (
+            patch.object(Transaction, 'verify_pow', verify_pow_wrapped),
+            patch.object(Transaction, 'verify_number_of_inputs', verify_number_of_inputs_wrapped),
+            patch.object(Transaction, 'verify_outputs', verify_outputs_wrapped),
```

Comment:
It's missing a wrapped call to the `verify_number_of_outputs()` which is usually called by `verify_outputs()`.

Reply:
Done in https://github.com/HathorNetwork/hathor-core/pull/824/commits/00116efab6ba074ef8aad62a3be8198fdfd6d4e5

Reply:
Thanks!